Scoop/NetworkManager.swift

```diff
@@ -10,15 +10,120 @@ import Alamofire
 
 class NetworkManager {
```

Comment:
You seem to be reusing a lot of the same code over and over. Can you find a way to refactor these functions? Maybe use generics for the return type?